kornia/augmentation/functional.py

```diff
@@ -0,0 +1,342 @@
+from typing import Tuple, List, Union
+
+import torch
+import torch.nn as nn
+from torch.distributions import Uniform
+
+from kornia.geometry.transform.flips import hflip, vflip
+from kornia.color.adjust import AdjustBrightness, AdjustContrast, AdjustSaturation, AdjustHue
+from kornia.color.gray import rgb_to_grayscale
+from kornia.geometry import pi
+
+
+UnionType = Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]
+FloatUnionType = Union[torch.Tensor, float, Tuple[float, float], List[float]]
+
+
+def random_hflip(input: torch.Tensor, p: float = 0.5, return_transform: bool = False) -> UnionType:
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = get_random_p_params(batch_size, p=p)
+    return apply_hflip(input, params, return_transform)
```

Comment:
one question it comes to my mind, I understand the abstraction of the api, however, maybe we could follow the pattern below:
- core level -> `kornia.hflip` (exposed)
- augmentation level -> `kornia.random_hflip` (exposed)
  - this needs rand random parameters
    - call `_get_random_p_params`  and `_apply_hflip` which internally calls `kornia.hflip`

same for the other functions. Basically, we should have like a top level api, and some specifics for augmentation that uses random parameters generators. Maybe /cc  @fmassa could help with this

Reply:
In my original thoughts, I prefer to mark the ```random_*``` APIs as deprecated. I just rewrote them for passing the tests.

I would only like to preserve:
- The functional API, the F in your example, to do the augmentations in command.
- The torch module-like augmentations, for doing the augmentations on the fly.

Reply:
I don't think we are going to deprecate methods at this moment. In order to proceed with this refactoring we might need more consensus about the API. I understand the need to reproduce results and I think it's necessary but first we should have an initial version of the module as close as possible to torchvision interface. We are going to do a release this week including the methods left in the roadmap. As a second iteration, we can sketch out about how to generalise code. We could open an issue for that in order to have a clear Idea about the whole thing before starting to code and make major changes in the API.

Reply:
Well, we _may_ think about deprecating random_* stuff later. But I like that I can do things both functional or Module way now. And I like the way it is done in current PR: split functionality into "generate random" and "apply random". Both might be called either from Module or from functional.

Reply:
in that case, I would add extra functions that accept the generated parameters. Similar to what has been done here https://github.com/kornia/kornia/pull/399. And probably, in terms of reproducibility I would provide such methods as functionals.

Reply:
Right, it makes sense to make everything work in the first place. In that case, is it better to split the current functional.py into two separated files:
- functional.py, to include all the functional operations.
- param_gen.py, to include all the random param generators.